src/control/drpc/modules.go

```diff
@@ -25,30 +25,140 @@
 
 package drpc
 
-import "github.com/golang/protobuf/proto"
+import (
+	fmt "fmt"
+
+	"github.com/golang/protobuf/proto"
+)
 
 // #cgo CFLAGS: -I${SRCDIR}/../../include
 // #include <daos/drpc_modules.h>
 import "C"
 
+const moduleMethodOffset = 100
+
 const (
-	// ModuleSecurityAgent is the dRPC module for security tasks in the
-	// DAOS agent
+	// ModuleSecurityAgent is the dRPC module for security tasks in DAOS agent
 	ModuleSecurityAgent = C.DRPC_MODULE_SEC_AGENT
 	// ModuleMgmt is the dRPC module for management service tasks
 	ModuleMgmt = C.DRPC_MODULE_MGMT
 	// ModuleSrv is the dRPC module for tasks relating to server setup
 	ModuleSrv = C.DRPC_MODULE_SRV
-	// ModuleSecurity is the dRPC module for security tasks in the DAOS
-	// server
+	// ModuleSecurity is the dRPC module for security tasks in DAOS server
 	ModuleSecurity = C.DRPC_MODULE_SEC
 )
 
+type Method struct {
+	id     int32
+	module int32
+}
```

Comment:
I was a bit surprised by this... I thought we were talking about defining Method as an interface, e.g.
```go
type Method interface {
        ID() int32
        Module() int32
        String() string
}
```

That way you could use the type alias in the const lists and have access to all of the extra functionality without requiring a call to GetMethod().
And then you'd have the module-specific method types, e.g.
```go
type mgmtMethod int32

func (m mgmtMethod) Module() int32 {
        return ModuleMgmt
}

func (m mgmtMethod) ID() int32 {
        return int32(m)
}

func (m mgmtModule) String() string {
        var mName string
        if mName, found := map[mgmtMethod]string {
                MethodPingRank: "ping rank",
        }[m]; found {
                return mName
        }
        return fmt.Sprintf("%d:%d", m.Module(), m.ID())
}
```

Reply:
You could even expand the interface a bit to include a `Valid() bool` method which does the validation that the method is within the expected range. Then the validation for each set of methods is right there along with the definition rather than in a different file. When adding a new module and methods it will be obvious what needs to be done.